llama.cpp

```diff
@@ -1697,6 +1683,161 @@ static std::string llama_format_tensor_shape(const struct ggml_tensor * t) {
+        try_override(OT & target, const struct llama_model_kv_override *override) {
+            if (!override) {
+                return false;
+            }
+            if (override->tag != LLAMA_KV_OVERRIDE_BOOL) {
+                return false;
+            }
+            target = override->bool_value;
+            return true;
+        }
+
+        template<typename OT>
+        static typename std::enable_if<!std::is_same<OT, bool>::value && std::is_integral<OT>::value, bool>::type
+        try_override(OT & target, const struct llama_model_kv_override *override) {
+            if (!override) {
+                return false;
+            }
+            if (override->tag != LLAMA_KV_OVERRIDE_INT) {
+                return false;
+            }
+            if (override->int_value < 0 && !std::is_signed<T>::value) {
+                return false;
+            }
+            target = override->int_value;
+            return true;
+        }
+
+        template<typename OT>
+        static typename std::enable_if<std::is_floating_point<OT>::value, bool>::type
+        try_override(T & target, const struct llama_model_kv_override *override) {
+            if (!override) {
+                return false;
+            }
+            if (override->tag != LLAMA_KV_OVERRIDE_FLOAT) {
+                return false;
+            }
+            target = override->float_value;
+            return true;
+        }
+
+        static bool set(const gguf_context * ctx, const int k, T & target, const struct llama_model_kv_override *override = nullptr) {
+            if (try_override<T>(target, override)) {
+                return true;
+            }
+            target = get_kv(ctx, k);
+            return true;
+        }
+
+        template <typename TT>
+        static bool set(const gguf_context * ctx, const char * key, TT & target, const struct llama_model_kv_override *override = nullptr) {
+            const int kid = gguf_find_key(ctx, key);
+            if (kid < 0) {
+                return false;
+            }
+            return GKV<TT>::set(ctx, kid, target, override);
+        }
+
+        template <typename TT>
+        static bool set(const gguf_context * ctx, const std::string & key, TT & target, const struct llama_model_kv_override *override = nullptr) {
+            return GKV<TT>::set(ctx, key.c_str(), target, override);
+        }
+    };
+
+    template<>
+    class GKV<std::string>: public GKV_Base<const char *> {
+        using BT = const char *;
+        public:
+        static bool set(const gguf_context * ctx, const int k, std::string & target, const struct llama_model_kv_override *override = nullptr) {
+            (void)override;
+            target = std::string(GKV<BT>::get_kv(ctx, k));
+            return true;
+        }
+
+        static bool set(const gguf_context * ctx, const char * key, std::string & target, const struct llama_model_kv_override *override = nullptr) {
```

Comment:
Having to write forwarding methods is also sort of annoying as well, so there's probably a better way.